.github/workflows/_checks-golang.yaml

```diff
@@ -50,21 +65,87 @@ jobs:
         with:
           submodules: true
           token: ${{ steps.app-token.outputs.token || github.token }}
+
       - name: Build
         run: |
           git config --global --add safe.directory "*"
           cd ${{ inputs.package_path }}
           make go-build
+
       - name: ModTidy check
         run: |
           cd ${{ inputs.package_path }}
           make go-mod-check
+
       - name: Lint check
         run: |
           cd ${{ inputs.package_path }}
           make go-lint-install
           make go-lint
+
       - name: Run tests
         run: |
           cd ${{ inputs.package_path }}
           make go-test
+
+      - name: Run coverage
+        if: inputs.enable_coverage
+        run: |
+          cd ${{ inputs.package_path }}
+          make go-coverage
+
+      - name: Upload coverage report
+        if: inputs.upload_coverage
+        uses: actions/upload-artifact@v4
+        with:
+          name: coverage-report-${{ github.run_number }}
+          path: ${{ inputs.package_path }}/profile.cov
+          retention-days: 30
+
+      - name: Quality Gate - Test coverage shall be above threshold
+        if: inputs.enable_coverage
+        shell: bash
+        env:
+          TESTCOVERAGE_THRESHOLD: ${{ inputs.coverage_threshold }}
+        run: |
```

Comment:
What if we post this message as a comment on the PR? I may be more useful than tracking action output. And it will be there for more time, as actions logs are pruned every N days on CI